### PR TITLE
Fix for IndexOutOfBounceException issue, and more

### PR DESCRIPTION
**Pull request by ideas-into-software**
- Fixed 'IndexOutOfBounceException' issue (https://github.com/geckoprojects-org/org.gecko.emf.utils/issues/17) - changed how object identifiers are obtained, previously using hash codes, however in large data sets duplicates occurred causing issues such as this one 

 - Upgraded to latest version of SODS library
 
 - Added logging

### CODE REVIEW

**Comment by ideas-into-software** [comment]
@juergen-albert

As briefly explained in commit notes, 'IndexOutOfBounceException' issue (https://github.com/geckoprojects-org/org.gecko.emf.utils/issues/17) was caused by duplicate hash codes - this occurrs in large data sets, causing issues such as this one. This was fixed by changing how EObject identifiers are obtained, i.e. instead of using `java.lang.Object.hashCode()`, now `org.eclipse.emf.ecore.util.EcoreUtil.getIdentification(EObject)` is used.

Regarding the other issue you reported (https://github.com/geckoprojects-org/org.gecko.emf.utils/issues/16) - based on profiling data collected, hot spots are present in SODS library / i.e. SODS library is where the bottleneck is; you can refer to attached export of profiler snapshots. Question is then: do you want me to spend time optimizing SODS library to try to improve its performance?

[jprofiler_20230825.zip](https://github.com/geckoprojects-org/org.gecko.emf.utils/files/12434535/jprofiler_20230825.zip)

File: org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java [hunk 1]
```diff
@@ -187,14 +200,16 @@ public void exportEObjectsTo(List<EObject> eObjects, OutputStream outputStream,
 		}
 	}
 
-	private void generatePseudoIDs(List<EObject> eObjects, Map<Integer, String> eObjectsPseudoIDs) {
-		final Set<Integer> processedEObjectsIdentifiers = new HashSet<Integer>();
+	private void generatePseudoIDs(List<EObject> eObjects, Map<String, String> eObjectsPseudoIDs) {
+		LOG.info("Generating pseudo IDs");
```
Comment (juergen-albert) [inline org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java:204]: I think this should be a debug log

File: org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java [hunk 2]
```diff
@@ -241,19 +257,21 @@ private String getID(EObject eObject) {
...
+	private String getPseudoID(EObject eObject, Map<String, String> eObjectsPseudoIDs) {
 		return eObjectsPseudoIDs.get(getEObjectIdentifier(eObject));
 	}
 
 	private void createSheets(SpreadSheet document, Map<String, Sheet> eClassesSheets, Set<EClass> eObjectsClasses,
-			Set<EEnum> eObjectsEnums, Map<Integer, String> eObjectsPseudoIDs, Map<Integer, Sheet> eObjectsSheets,
+			Set<EEnum> eObjectsEnums, Map<String, String> eObjectsPseudoIDs, Map<String, Sheet> eObjectsSheets,
 			List<EObject> eObjects, Map<Object, Object> exportOptions) {
+	
+		LOG.info("Creating sheets");
```
Comment (juergen-albert) [inline org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java:272]: I think this should be a debug log

File: org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java [hunk 3]
```diff
@@ -271,10 +289,12 @@ private void createSheets(SpreadSheet document, Map<String, Sheet> eClassesSheet
 	}
 
 	private void createSheetsData(SpreadSheet document, Map<String, Sheet> eClassesSheets, Set<EClass> eObjectsClasses,
-			Set<EEnum> eObjectsEnums, Map<Integer, String> eObjectsPseudoIDs, Map<Integer, Sheet> eObjectsSheets,
+			Set<EEnum> eObjectsEnums, Map<String, String> eObjectsPseudoIDs, Map<String, Sheet> eObjectsSheets,
 			List<EObject> eObjects, Map<Object, Object> exportOptions) throws EMFExportException {
 
-		final Set<Integer> processedEObjectsIdentifiers = new HashSet<Integer>();
+		LOG.info("Creating sheets' data");
```
Comment (juergen-albert) [inline org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java:295]: I think this should be a debug log

File: org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java [hunk 4]
```diff
@@ -968,6 +1005,9 @@ private Sheet addMetadataSheet(SpreadSheet document, EEnum eEnum) {
 
 	private void exportMetadata(SpreadSheet document, Set<EClass> eClasses, Set<EEnum> eEnums,
 			Map<Object, Object> exportOptions) {
+
+		LOG.info("Exporting metadata");
```
Comment (juergen-albert) [inline org.gecko.emf.exporter.ods/src/org/gecko/emf/exporter/ods/EMFODSExporter.java:1009]: I think this should be a debug log

**Pull request merged**